### ISSUE TITLE
User Space breaking bug in Linux - Global Appmenu, Vala

### ISSUE THREAD

**Issue by rbreaves** [microsoft/vscode#125012]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.56.2 054a9295330880ed74ceaedda236253b4f39a335 x64
- OS Version: Ubuntu Budgie 20.04
Desktop Environment: budgie
DE Version: 10.5.1-6ubuntu0.2

Steps to Reproduce:

1. Open up VS Code
2. Open up literally any folder on your system
3. Witness the crash

This happens so quickly and dumps the user completely out of their session that it doesn't even log much of anything when the crash reporter is running, no dump file. Just a client-id file with a single hash. That is it. If a better suggestion for what I can run can be provided then I will do so, but what is suggested here https://github.com/microsoft/vscode/wiki/Native-Crash-Issues does not work for me.

XFCE4 handles this a little bit better, by handling the exception and allowing the user to restart the panel.
![120095420-879d1b80-c0eb-11eb-8697-e8a1b3e8a1d3](https://user-images.githubusercontent.com/10969616/120250387-8da61000-c243-11eb-9bf5-6e7b9804031a.png)

Ideally whatever is going on with the menu that causes Vala-appmenu to crash unexpectedly would not happen. Even more ideally Budgie would better handle an exception created by either VS Code or Vala-appmenu. My hope is that I can get this issue resolved and possibly in multiple places, layers and levels to help increase overall stability.

Logs from Budgie to help give a better idea of the nature of the problem.
```
** (budgie-panel:389626): WARNING **: 20:02:56.644: GDBus.Error:org.gtk.GDBus.UnmappedGError.Quark._LIBDBUSMENU_2dGLIB.Code0: The ID supplied 206 does not refer to a menu item we have
** (budgie-panel:389626): WARNING **: 20:02:56.942: GDBus.Error:org.gtk.GDBus.UnmappedGError.Quark._LIBDBUSMENU_2dGLIB.Code0: The ID supplied 213 does not refer to a menu item we have
** (budgie-panel:389626): WARNING **: 20:02:56.942: GDBus.Error:org.gtk.GDBus.UnmappedGError.Quark._LIBDBUSMENU_2dGLIB.Code0: The ID supplied 223 does not refer to a menu item we have
[1]  + 389626 segmentation fault (core dumped)  budgie-panel --replace
```

Additional references
https://gitlab.com/vala-panel-project/vala-panel-appmenu/-/issues/347
https://github.com/solus-project/budgie-desktop/issues/2112
https://github.com/UbuntuBudgie/budgie-desktop/issues/68

**Comment by rbreaves** [comment]
I believe this issue was resolved by the vala-appmenu author in his 0.7.6 release, as well as 0.7.5, so the repos for 20.04 either need to be updated to the latest or users may need to manually locate 0.7.5 or 0.7.6 deb files and install them manually. He responded to the post I made there and was unable to duplicate this behavior on the latest.

**Comment by rbreaves** [comment]
Latest dev patch by @fossfreedom seems to have fixed this issue in Ubuntu Budgie 20.04.
https://github.com/UbuntuBudgie/budgie-desktop/issues/68
